hapi/base.py

```diff
@@ -112,7 +118,7 @@ def _execute_request_raw(self, conn, request):
         except:
             raise HapiTimeout(None, request, traceback.format_exc())
 
-        encoding = [i[1] for i in result.getheaders() if i[0] == 'content-encoding']
+        encoding = [i[1] for i in result.getheaders() if i[0].lower() == 'content-encoding']
```

Comment:
```('content-encoding', 'gzip')``` in python 2, ```('Content-Encoding', 'gzip')``` in python 3. Convert it to lowercase for python2/3 compatibility here.